VirtualPetAdoptionCenter.Web/Pages/PetActivity.cshtml.cs

```diff
@@ -0,0 +1,38 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using VirtualPetAdoptionCenter.Core.Services;
+using VirtualPetAdoptionCenter.Models.DomainModels;
+
+namespace VirtualPetAdoptionCenter.Web.Pages
+{
+    public class PetActivityModel : PageModel
```

Comment:
The model handles both data fetching and grooming time updates. You could separate these responsibilities into different classes or services.